devito/passes/clusters/blocking.py

```diff
@@ -161,3 +181,96 @@ def decompose(ispace, d, block_dims):
     directions.update({bd: ispace.directions[d] for bd in block_dims})
 
     return IterationSpace(intervals, sub_iterators, directions)
+
+
+class Skewing(Queue):
+
+    """
+    Rebuild clusters with skewed expressions and skewed IterationSpace.
+
+    Notes
+    -----
+    This transformation is applying loop skewing to derive the
+    wavefront method of execution of nested loops. Loop skewing is
+    a simple transformation of loop bounds and is combined with loop
+    interchanging to generate the wavefront [1]_.
+
+    .. [1] Wolfe, Michael. "Loops skewing: The wavefront method revisited."
+    International Journal of Parallel Programming 15.4 (1986): 279-293.
+
+    Examples:
+
+    .. code-block:: python
+
+        for i = 2, n-1
+            for j = 2, m-1
+                a[i,j] = (a[a-1,j] + a[i,j-1] + a[i+1,j] + a[i,j+1]) / 4
+
+    to
+
+    .. code-block:: python
+
+        for i = 2, n-1
+            for j = 2+i, m-1+i
+                a[i,j-i] = (a[a-1,j-i] + a[i,j-1-i] + a[i+1,j-i] + a[i,j+1-i]) / 4
+
+    """
+
+    def __init__(self, options):
+        self.skewinner = bool(options['skewinner'])
+
+        super(Skewing, self).__init__()
+
+    def callback(self, clusters, prefix):
+
+        if not prefix:
+            return clusters
+
+        d = prefix[-1].dim
+
+        processed = []
+        for c in clusters:
+            # Explore skewing possibilities only in case dimension d is parallel,
+            # not innermost and is not incrementing by a Symbol. Symbol increments
+            # are encountered in blocked (tiled) loops where symbols are used as
+            # increments to help parametrize the block shape.
+            if (PARALLEL not in c.properties[d] or d.symbolic_incr.is_Symbol):
+                return clusters
+
+            if d is c.ispace[-1].dim and self.skewinner is False:
+                return clusters
+
+            mapper, intervals = {}, []
+            skew_dim = None
+
+            relations = c.ispace.intervals.relations
+
+            properties = {k: v for k, v in c.properties.items()}
+            for n, i in enumerate(c.ispace):
+                # Identify a skew_dim and its position
+                if (SEQUENTIAL in c.properties[i.dim] and
+                   not i.dim.symbolic_incr.is_Symbol):
```

Comment:
Remind me what is `not i.dim.symbolic_incr.is_Symbol` for?

Reply:
Actually, this is a remainder from a blocked time loop. I drop it now since it not needed for this PR, but will definitely need to take some care later...